src/crosswalk/plots.py

```diff
@@ -0,0 +1,374 @@
+    plt.fill_between(pred_df[dose_variable], y_lo_fe, y_hi_fe, 
+        alpha=0.75, color='darkgrey')
+    plt.plot(pred_df[dose_variable], y_mean, color='black', linewidth=0.75)
+    plt.xlim([min_cov, max_cov])
+    if ylim is not None:
+        plt.ylim(ylim)
+    plt.xlabel('Exposure', fontsize=10)
+    plt.xticks(fontsize=10)
+    plt.ylabel('Effect size', fontsize=10)
+    plt.yticks(fontsize=10)
+
+    # other comparison
+    non_direct_df = data_df.loc[
+    (data_df.dorm_ref != cwmodel.gold_dorm) | (data_df.dorm_alt != obs_method)
+    ]
+    # direct comparison
+    plot_data_df = data_df.loc[
+    (data_df.dorm_ref == cwmodel.gold_dorm) & (data_df.dorm_alt == obs_method)
+    ]
+    
+    for key, value in plot_key.items():
+        plt.scatter(
+            plot_data_df.loc[plot_data_df.plot_guide == key, f'{dose_variable}'],
+            plot_data_df.loc[plot_data_df.plot_guide == key, 'y'],
+            s=plot_data_df.loc[plot_data_df.plot_guide == key, 'size_var'],
+            marker=value[0], facecolors=value[1], edgecolors=value[2], 
+            linewidth=0.6, alpha=.6, label=key
+        )
+    plt.scatter(non_direct_df[f'{dose_variable}'], non_direct_df['y'],
+                facecolors='grey', edgecolors='grey', alpha=.3,
+                s=non_direct_df.loc[non_direct_df.plot_guide == key, 'size_var'])
+    # Content string with betas
+    betas = list(np.round(cwmodel.fixed_vars[obs_method], 3))
+    content_string = ""
+    for idx in np.arange(len(cwmodel.cov_models)):
+        cov = cwmodel.cov_models[idx].cov_name
+        knots_slices = lst_slices[idx]
+        content_string += f"{cov}: {betas[knots_slices]}; "
+    # Plot title
+    if plot_note is not None:
+        plt.title(content_string, fontsize=10)
+        plt.suptitle(plot_note, y=1.01, fontsize=12)
+    else:
+        plt.title(content_string, fontsize=10)
+    plt.legend(loc='upper left')
+    
+    for knot in knots:
+        plt.axvline(knot, color='navy', linestyle='--', alpha=0.5, linewidth=0.75)
+    # Save plots
+    if write_file:
+        assert cwdir is not None, "cwdir is not specified!"
+        outfile = os.path.join(cwdir, f'{file_name}.pdf')
+        plt.savefig(outfile, orientation='landscape', bbox_inches='tight')
+        print(f"Dose response plot saved at {outfile}")
+    else:
+        plt.show()
+
+
+def funnel_plot(obs_method='Self-reported', cwdata=None, cwmodel=None, 
+                continuous_variables=[], cwdir=None, file_name='funnel_plot', 
+                plot_note=None, include_bias=False, write_file=False):
+    """Funnel Plot.
+    Args:
+        obs_method (str):
+            Alternative definition or method intended to be plotted.
+        cwdata (CWData object):
+            CrossWalk data object.
+        cwmodel (CWModel object):
+            Fitted CrossWalk model object.
+        continuous_variables (list):
+            List of continuous covariate names.
+        cwdir (str):
```

Comment:
Will be nice to change `cwdir` to `result_dir` so that it is more informative.

Reply:
i thought about that, or plots_dir? which do you prefer?

Reply:
Both are good for me, better than `cwdir` :)